bot/utils/parameter_value_generator.py

```diff
@@ -17,42 +16,56 @@ def __call__(self, *args):
 
 
 def parameter_value_generator(
-    chemistry,
-    parameter,
-    lower_bound=None,
-    upper_bound=None,
+    params,
+    parameter_dict,
 ):
     """
     Generates random values for a given parameter and
     for a given chemistry.
     Parameters:
-        chemistry: dict
-        parameter: str
-        lower_bound: numerical
-        upper_bound: numerical
+        params: pybamm.ParameterValues
+        parameter_dict: dict
+            Parameters to be varied. Should be of the form -
+            {
+                "parameter1": {
+                    "lower bound": numerical or None,
+                    "upper bound": numerical or None
+                },
+                "parameter2": {
+                    "lower bound": numerical or None,
+                    "upper bound": numerical or None
+                },
+            }
     Returns:
         params: pybamm.ParameterValues
-        new_parameter_value: numerical
     """
-    params = pybamm.ParameterValues(chemistry=chemistry)
-
-    if callable(params[parameter]):
-        base_value = 1
-        new_parameter_value = desired_decimal_point_generator(
-            lower_bound if lower_bound is not None else base_value*0.5,
-            upper_bound if upper_bound is not None else base_value*2,
-            2
-        )
-        params[parameter] = FunctionLike(
-            params[parameter], new_parameter_value
-        )
-    else:
-        base_value = params[parameter]
-        new_parameter_value = desired_decimal_point_generator(
-            lower_bound if lower_bound is not None else base_value*0.5,
-            upper_bound if upper_bound is not None else base_value*2,
-            2
-        )
-        params[parameter] = new_parameter_value
-
-    return params, new_parameter_value
+
+    for parameter in parameter_dict.keys():
```

Comment:
you could do `for parameter, bounds in parameter_dict.items():` to avoid indexing in to the dict as much

Reply:
Oh yes, that would be much cleaner